### ISSUE TITLE
Figure out how to calculate image size

### ISSUE THREAD

**Issue by jzt** [vmware/vic#975]
Since docker has removed layer `Size` attribute from the `v1Compatibility` metadata, we need a way to calculate image size. My first attempt was to calculate each layer size in `imagec/docker.go` during the step that calculates the `diffID`:

```go
	// Copy bytes from decompressed layer into diffIDSum to calculate diffID
	bytesRead, cerr := io.Copy(diffIDSum, tar)
	if cerr != nil {
		return diffID, cerr
	}
```

If the layer being processed represents an empty layer (meaning it has not changed the filesystem at all and thus has no size), I skip it. Otherwise, `bytesRead` becomes the size of the layer.

```go
	if diffID != string(DigestSHA256EmptyTar) {
		// bytesRead represents the size of the decompressed layer data
		image.size = bytesRead
	}
```

However, when summing up the layer sizes in `imagec/imagec.go:CreateImageConfig()`, it results in an incorrect image size:

```go
func CreateImageConfig(images []*ImageWithMeta, manifest *Manifest) error {

	imageLayer := images[0] // the layer that represents the actual image
	image := docker.V1Image{}
	rootFS := docker.NewRootFS()
	history := make([]docker.History, 0, len(images))
	diffIDs := make(map[string]string)
	var size int64

	// step through layers to get command history and diffID from oldest to newest
	for i := len(images) - 1; i >= 0; i-- {
		layer := images[i]
		if err := json.Unmarshal([]byte(layer.meta), &image); err != nil {
			return fmt.Errorf("Failed to unmarshall layer history: %s", err)
		}
		h := docker.History{
			Created:   image.Created,
			Author:    image.Author,
			CreatedBy: strings.Join(image.ContainerConfig.Cmd, " "),
			Comment:   image.Comment,
		}
		history = append(history, h)
		rootFS.DiffIDs = append(rootFS.DiffIDs, dockerLayer.DiffID(layer.diffID))
		diffIDs[layer.diffID] = layer.ID
		size += layer.size
	}
```

Expected:
```
$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
hello-world         latest              94df4f0ce8a4        5 weeks ago         967 B
busybox             latest              47bcc53f74dc        11 weeks ago        1.113 MB
```

Actual:
```
$ docker -H 192.168.218.219:2376 --tls --tlscert='./vic-docker-cert.pem' --tlskey='./vic-docker-key.pem' images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
hello-world         latest              94df4f0ce8a4        5 weeks ago         2.56 kB
busybox             latest              47bcc53f74dc        11 weeks ago        1.311 MB
```

**Comment by jzt** [comment]
I tried unmarshalling into an `ImageConfig` struct that is allocated at the top of the for loop each iteration, but its still not right.

**Comment by jzt** [comment]
Closing as this was fixed in #977.
